### PR TITLE
Lift meson cap, remove conda-build work-arounds

### PR DESCRIPTION
**Pull request by h-vetinari**
For https://github.com/mesonbuild/meson/issues/13403

### CODE REVIEW

**Comment by h-vetinari** [comment]
@conda-forge/matplotlib 
This removes a couple of work-arounds now that the respective fixes have landed in meson & conda-build. It also switches over the build invocation to go through meson-python, as invoking meson directly and then meson-python is not a supported operation. PTAL :)

File: recipe/build_base.sh [hunk 1]
```diff
@@ -8,9 +7,14 @@ export AR=$GCC_AR
...
 export MESON_ARGS="${MESON_ARGS} -Dsystem-freetype=true -Dsystem-qhull=true"
 
-mkdir builddir
-$PYTHON -m mesonbuild.mesonmain setup builddir $MESON_ARGS
-cat builddir/meson-logs/meson-log.txt
-$PYTHON -m build --wheel \
-        --no-isolation --skip-dependency-check -Cbuilddir=builddir
+# meson-python already sets up a -Dbuildtype=release argument to meson, so
+# we need to strip --buildtype out of MESON_ARGS or fail due to redundancy
+MESON_ARGS_REDUCED="$(echo $MESON_ARGS | sed 's/--buildtype release //g')"
```
Comment (jakirkham) [inline recipe/build_base.sh:12]: Do we have an issue tracking removing this flag from `MESON_ARGS` upstream?

Comment (h-vetinari) [reply recipe/build_base.sh:12]: No, it's their default. Our compilers need to change, and they did as of [clang 17](https://github.com/conda-forge/clang-compiler-activation-feedstock/pull/112) / [GCC 13](https://github.com/conda-forge/clang-compiler-activation-feedstock/pull/112)

Comment (jakirkham) [reply recipe/build_base.sh:12]: Sorry I meant raise a conda-forge issue, but it sounds like that already was fixed. Do you have a link?

Comment (h-vetinari) [reply recipe/build_base.sh:12]: There are two links in my comment...? 🤔

Comment (jakirkham) [reply recipe/build_base.sh:12]: Oh oops. Sorry was on my phone, which didn't make this clear. Thanks!

**Pull request merged**